### PR TITLE
Address management chart

### PR DESCRIPTION
**Pull request by salman-neslit**
## Link to Issue
Closes: #8709

## Description of Changes
- Introduced a "Manage Wallet Addresses" section on the edit profile page.
- Added an interactive table displaying wallet addresses and their associated communities.
- Enabled functionality to add and remove wallet addresses.
- Implemented "..." menu with additional options for each wallet address.
- Dynamic table expansion for multiple wallet addresses and connected communities.

## Test Plan
- Navigate to the Edit Profile page.
- Go to the new "Manage Addresses" section.
- Verify the following:
  - The new UI is displayed correctly with wallet addresses on the left and associated communities on the right.
  - The "..." menu appears for each wallet address and shows the disconnect options when clicked.
  - Removing a wallet address:
    -  Click on the "..." menu for a wallet address.
    - Select the "Disconnect Address" option.
    - Confirm that the wallet address is removed from the table.

## Other Considerations

Figma Link : https://www.figma.com/design/N7vGFBW0XNQG5lgUIflObX/Address-Mgmt?node-id=223-12578&t=TO2EMFwyebMRpS74-0

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityInfo/CWCommunityInfo.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWCommunityInfo/CWCommunityInfo.tsx:1]: please reuse/modify `packages/commonwealth/client/scripts/views/components/component_kit/CommunityInfo/CommunityInfo.tsx`

**Review by masvelio — COMMENTED** [review]
1. You used vertical 3 dots (which is too big imo btw) instead of horizontal. Check 3 dots button in thread or comment

image

2. Cursor pointer indicates that element is clickable. So either use `navigateToCommunity` on click or change the cursor


https://github.com/user-attachments/assets/ccbed81e-bcbc-44f8-8139-29b823582693

File: packages/commonwealth/client/scripts/views/components/linked_addresses.tsx [hunk 1]
```diff
@@ -69,27 +101,53 @@ export const LinkedAddresses = (props: LinkedAddressesProps) => {
...
-            addressInfo={addr}
-            toggleRemoveModal={(val: boolean, address: AddressInfo) => {
-              setIsRemoveModalOpen(val);
-              setCurrentAddress(address);
-            }}
-          />
-        );
-      })}
+    <div>
+      {TableComponent}
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/linked_addresses.tsx:145]: let's just use CWTable here

**Review by masvelio — APPROVED** [review]
Approving in advance but please fix some of Malik's and mine comments

**Review by Israellund — APPROVED** [review]

**Review by mzparacha — APPROVED** [review]

**Comment by dillchen** [comment]
Hi @salman-neslit does this address also #8711 and #8710 if so could you please link them as related issues?

**Comment by salman-neslit** [comment]
> Hi @salman-neslit does this address also #8711 and #8710 if so could you please link them as related issues?

@dillchen  This PR addresses many aspects of issue [#8710](https://github.com/hicommonwealth/commonwealth/issues/8710) ; however, it does not cover one specific edge case associated with it.

**Comment by dillchen** [comment]
@salman-neslit okay that's fine. So we can merge this but there will be follow up tickets. Can you provide context on the edge case that's not covered?

File: packages/commonwealth/client/scripts/views/components/linked_addresses.tsx (hunk 1 above)
Comment (salman-neslit) [reply packages/commonwealth/client/scripts/views/components/linked_addresses.tsx:145]: useMemo is used to memoize the CWTable component. This optimisation prevents unnecessary recalculations of the table structure and improves  performance.

```
  const TableComponent = useMemo(() => {
    return <CWTable columnInfo={columnInfo} rowData={rowData} />;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [addresses]);
```

**Pull request merged**